### PR TITLE
chore(label cleaning): remove windows and add jdk 17

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3223#issuecomment-1323832007
removing windows agent label and adding jdk17

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -13,11 +13,11 @@ properties([
 // Define the sequential stages and the parallel steps inside each stage
 def sequentialStages = [:]
 sequentialStages['Tool'] = [ 'java', 'maven', 'maven-11', 'maven-windows', 'maven-11-windows']
-sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'windows']
+sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'jdk17'] // Remove windows label as removed and add jdk17 as used
```
Comment (dduportal) [inline Jenkinsfile]:
```suggestion
sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'jdk17'] // Label definition found here: https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L45-L65
```

not sure to understand why adding the comment and its meaning, so I propose to comment with the link to the label definition.

Comment (dduportal) [reply Jenkinsfile]: Second though: I'm not sure if it make sense to test for the labels `jdk8`, etc. 
What are the pipeline or library consuming these labels? (question for both @smerle33 and @MarkEWaite )

Comment (smerle33) [reply Jenkinsfile]: > Second though: I'm not sure if it make sense to test for the labels `jdk8`, etc. What are the pipeline or library consuming these labels? (question for both @smerle33 and @MarkEWaite )

I don't see any reference to jdk8 within pipeline-library but not sure that it's the main reference. Shouldn't we use the ci logs to ensure the list for agent currently used ?
from the code in pipeline-library I can only see : 
- maven-11
- docker-windows
- docker || linux-amd64-docker
- maven
- maven-(+jdk)(+-windows) map
- vm && linux
- linux

not quite sure for :
- jnlp

Comment (dduportal) [reply Jenkinsfile]: we have to search the whole pipeline-library repository, but also the whole jenkinsci and jenkins-infra.

I totally challenge the fact that these labels (`jdk8`, `jdk11`, `jdk17`) are tested unless we know for sure where there are used.

----

`jnlp` could have been a former "default". I don't think we should test it either (to be searched as well).

File: Jenkinsfile [hunk 2]
```diff
@@ -13,11 +13,11 @@ properties([
 // Define the sequential stages and the parallel steps inside each stage
 def sequentialStages = [:]
 sequentialStages['Tool'] = [ 'java', 'maven', 'maven-11', 'maven-windows', 'maven-11-windows']
-sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'windows']
+sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'jdk17'] // Label definition found here: https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L45-L65
```
Comment (dduportal) [inline Jenkinsfile]:
```suggestion
```

I would delete this line and add the `vm && linux` in the line above

File: Jenkinsfile [hunk 3]
```diff
@@ -13,11 +13,11 @@ properties([
 // Define the sequential stages and the parallel steps inside each stage
 def sequentialStages = [:]
 sequentialStages['Tool'] = [ 'java', 'maven', 'maven-11', 'maven-windows', 'maven-11-windows']
```
Comment (dduportal) [inline Jenkinsfile]:
```suggestion
// Labels requested in https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy and https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPluginWithGradle.groovy
sequentialStages['Tool'] = [ 'maven', 'maven-11', 'maven-17', 'maven-19', 'maven-windows', 'maven-11-windows', 'maven-17-windows', 'maven-19-windows']
```

File: Jenkinsfile [hunk 4]
```diff
@@ -13,11 +13,11 @@ properties([
...
-sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'windows']
+sequentialStages['OS & Java'] = [ 'linux', 'jdk8', 'jdk11', 'jdk17'] // Label definition found here: https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L45-L65
 sequentialStages['Processor'] = [ 'arm64', 'amd64' ] // Remove ppc64le and s390x until virtual machine available 'ppc64le', 's390x'
 sequentialStages['Docker'] = [ 'arm64docker', 'docker', 'docker-windows'] // Remove ppc64le and s390x until available again 'ppc64ledocker', 's390xdocker'
 sequentialStages['Memory'] = [ 'highmem', 'highram']
-sequentialStages['Cloud & Orchestrator'] = [ 'aci', 'aws', 'azure', 'kubernetes']
+sequentialStages['Cloud & Orchestrator'] = [ 'aci', 'aws', 'azure', 'kubernetes', 'do']
 sequentialStages['JDK'] = [ 'maven-8', 'maven-11', 'maven-17']
```
Comment (dduportal) [inline Jenkinsfile]:
```suggestion
```

**Pull request merged**